x-pack/plugins/apm/server/routes/services/route.ts

```diff
@@ -252,13 +253,35 @@ const serviceMetadataDetailsRoute = createApmServerRoute({
       kuery: '',
     });
 
-    return getServiceMetadataDetails({
+    const serviceMetadataDetails = await getServiceMetadataDetails({
       serviceName,
       setup,
       searchAggregatedTransactions,
       start,
       end,
     });
+
+    if (serviceMetadataDetails?.container?.id) {
```

Comment:
note: Send additional request for metrics data when clicking on the popup seems a cleaner approach .